### PR TITLE
XRT Device changes in XMA code

### PR DESCRIPTION
**Pull request by vboggara-xilinx**
#### Problem solved by the commit
moved xclDeviceHandle to xrt::device in XMA code base.

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
N/A

#### How problem was solved, alternative solutions (if any) and why they were rejected
Its PR related work, https://jira.xilinx.com/browse/VITIS-3278

#### Risks (if any) associated the changes in the commit
Low Risk

#### What has been tested and how, request additional testing if necessary
Ran XMA sprite testcases locally with this change, all are passing without any issue

#### Documentation impact (if any)
N/A

### CODE REVIEW

**Comment by vboggara-xilinx** [comment]
Migrating XMA code base with XRT native infra, In the PR made changes related to xrt::device.

File: src/xma/src/xmaapi/xmahw_hal.cpp [hunk 1]
```diff
@@ -115,33 +104,24 @@ bool hal_configure(XmaHwCfg *hwcfg, XmaXclbinParameter *devXclbins, int32_t num_
         }
 
         hwcfg->devices.emplace_back(XmaHwDevice{});
-
         XmaHwDevice& dev_tmp1 = hwcfg->devices.back();
-
-        dev_tmp1.handle = xclOpen(dev_index, NULL, XCL_QUIET);
-        if (dev_tmp1.handle == NULL){
+        dev_tmp1.xrt_device = xrt::device(dev_index);
+        if (dev_tmp1.xrt_device.get_handle()->get_device_handle() == NULL){
```
Comment (stsoe) [inline src/xma/src/xmaapi/xmahw_hal.cpp:109]: nullptr

**Review by stsoe — APPROVED** [review]
Thanks, this lgtm.   

For other volunteer reviewers who might look, please these are small evolutionary steps towards a grander goal, so hold your horses on changes other what involves new code here, we will get to it all sooner or later.

**Review by chvamshi-xilinx — APPROVED** [review]
looks good

File: src/xma/src/xmaapi/xmahw_hal.cpp (hunk 1 above)
Comment (vboggara-xilinx) [reply src/xma/src/xmaapi/xmahw_hal.cpp]: @stsoe , addressed your comment.

**Comment by stsoe** [comment]
retest this please - unrelated failure another timeout issue and job got killed

**Comment by salindac** [comment]
retest this please - unrelated failure another timeout issue and job got killed

**Review by sarab96 — APPROVED** [review]
Looks good

**Pull request merged**